apps/system/init/init.c

```diff
@@ -157,5 +166,16 @@ int preapp_start(int argc, char *argv[])
 	}
 #endif
 
+/***********************************************************************************
+ *	current preapp_start does the up_cxxinitialize which initializes the
+ *	static constructors. All the tasks and threads created by preapp refer to
+ *	these constructors. To make sure that c++ test cases from tash runs properly,
+ *	we need to keep preapp/constructor initializations alive with below changes.
+ **********************************************************************************/
+#ifdef CONFIG_HAVE_CXXINITIALIZE
+	while (1) {
```

Comment:
While loop is not required here as semaphore is not available at all, wait on a signal is even better.

Reply:
done